### PR TITLE
Support auto delete setting in az ml data

### PR DESCRIPTION
**Pull request by iuyoy**
# Description
To support auto delete setting in az ml data command.

Spec: https://github.com/Azure/azureml_run_specification/blob/master/specs/data-asset-lifecycle-mgmt.md

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by pepelu0 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/asset.py [hunk 1]
```diff
@@ -21,6 +22,7 @@ class AssetSchema(ResourceSchema):
     version = VersionField()
     creation_context = NestedField(CreationContextSchema, dump_only=True)
     latest_version = fields.Str(dump_only=True)
+    auto_delete_setting = ExperimentalField(NestedField(AutoDeleteSettingSchema))
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/asset.py:25]: Does this applly to all asset types?  Or it's just data asset?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/asset.py [hunk 2]
```diff
@@ -26,6 +26,8 @@ class Asset(Resource):
     :type tags: dict[str, str]
     :param properties: The asset property dictionary.
     :type properties: dict[str, str]
+    :param properties: The asset property dictionary.
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/asset.py:29]: update

File: sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_asset_utils.py [hunk 3]
```diff
@@ -185,3 +194,34 @@ def test_directory_size_with_ignore_file(self, storage_test_directory: str, amli
         _, _ = generate_link_file(storage_test_directory)
         with_symlink_size = get_directory_size(storage_test_directory)
         assert len(with_symlink_size[1]) == len(base_size[1]) + 2
+
+    def test_check_or_modify_auto_delete_setting(self):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_asset_utils.py:198]: add some e2e test for this feature?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/auto_delete_setting.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+from typing import Any
+from azure.ai.ml._utils._experimental import experimental
+from azure.ai.ml.constants._common import AutoDeleteCondition
+from azure.ai.ml.entities._mixins import RestTranslatableMixin
+from azure.ai.ml._restclient.v2023_04_01_preview.models import AutoDeleteSetting as RestAutoDeleteSetting
+
+
+@experimental
+class AutoDeleteSetting(RestTranslatableMixin):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/auto_delete_setting.py:13]: Need to make this a public class

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/asset.py (hunk 1 above)
Comment (iuyoy) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/asset.py:25]: Auto delete setting should be a base property for all assets, but only be used in data asset at first.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/asset.py (hunk 2 above)
Comment (iuyoy) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/asset.py]: Removed

File: sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_asset_utils.py (hunk 3 above)
Comment (iuyoy) [reply sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_asset_utils.py:198]: add a e2e test on sdk/ml/azure-ai-ml/tests/dataset/e2etests/test_data.py

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/auto_delete_setting.py (hunk 4 above)
Comment (iuyoy) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/auto_delete_setting.py]: It is only used for CLI command

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**